packages/button/src/Button.tsx

```diff
@@ -1,483 +1,182 @@
-      box-shadow: inset 0 2px 2px ${uiColors.green.dark2};
-    }
-
-    &:focus,
-    &:hover {
-      color: ${uiColors.white};
-    }
-  `,
-
-  [Variant.Info]: css`
-    color: ${uiColors.green.base};
-    background-color: transparent;
-    background-image: none;
-    border: 1px solid ${uiColors.green.base};
-    box-shadow: none;
-
-    &:before {
-      background-color: #129f4c;
-      background-image: linear-gradient(#129f4c, #148040);
-      box-shadow: 0 1px 4px ${transparentize(0.9, uiColors.black)},
-        inset 0 -1px 0 #158242;
-    }
-
-    &:after {
-      background-color: ${uiColors.green.base};
-      background-image: linear-gradient(#148040, #129f4c);
-      box-shadow: inset 0 2px 2px ${uiColors.green.dark2};
-    }
-
-    &:not(:disabled) {
-      &:focus,
-      &:hover,
-      &:active {
-        border-color: #158242;
-        color: ${uiColors.white};
-      }
-    }
-  `,
-
-  [Variant.Danger]: css`
-    color: ${uiColors.white};
-    border: 1px solid ${uiColors.red.dark2};
-    background-color: ${uiColors.red.base};
-    background-image: linear-gradient(#e45b26, #b63016);
-    box-shadow: inset 0 -1px 0 0 ${uiColors.red.dark2};
+import { Variant, Size, ButtonProps } from './types';
+import { getClassName, getIconStyle } from './styles';
 
-    &:before {
-      background-color: ${uiColors.red.dark2};
-      background-image: linear-gradient(#e45b26, ${uiColors.red.dark2});
-      box-shadow: 0 1px 4px ${transparentize(0.9, uiColors.black)},
-        inset 0 -1px 0 ${uiColors.red.dark2};
-    }
-
-    &:after {
-      background-color: ${uiColors.red.dark2};
-      background-image: linear-gradient(#ad231b, #e45b26);
-      box-shadow: inset 0 2px 2px ${uiColors.red.dark2};
-    }
-
-    &:focus,
-    &:hover {
-      color: ${uiColors.white};
-    }
-  `,
-
-  [Variant.Dark]: css`
-    color: ${uiColors.white};
-    border: 1px solid ${uiColors.gray.dark2};
-    background-color: ${uiColors.gray.dark1};
-    background-image: linear-gradient(
-      ${uiColors.gray.base},
-      ${uiColors.gray.dark1}
-    );
-    box-shadow: inset 0 -1px 0 ${uiColors.gray.dark2};
-
-    &:before {
-      background-image: linear-gradient(
-        ${uiColors.gray.base},
-        ${uiColors.gray.dark2}
-      );
-      box-shadow: 0 1px 4px ${transparentize(0.9, uiColors.black)},
-        inset 0 -1px 0 ${uiColors.gray.dark2};
-    }
-
-    &:after {
-      background-image: linear-gradient(
-        ${uiColors.gray.dark1},
-        ${uiColors.gray.base}
-      );
-      box-shadow: inset 0 2px 2px ${uiColors.gray.dark2};
-    }
-
-    &:focus,
-    &:hover {
-      color: ${uiColors.white};
-    }
-  `,
-};
-
-const interactionRingSizes: Record<Size, string> = {
-  [Size.XSmall]: css`
-    height: 22px;
-  `,
-
-  [Size.Small]: css`
-    height: 25px;
-  `,
-
-  [Size.Normal]: css`
-    height: 32px;
-  `,
+const rippleStyle = css`
+  overflow: hidden;
+  border-radius: 3px;
+  flex-grow: 1;
```

Comment:
Since this is position absolute, I don't think the flex grow property is doing anything here.

Reply:
👍